src/util/head.ts

```diff
@@ -1,4 +1,17 @@
+import SimplePath from '../@types/SimplePath'
+import ThoughtId from '../@types/ThoughtId'
+
+/**
+ * Note: We add an overload for `SimplePath` here to avoid TypeScript inferring `string | ThoughtId`
+ * as part of the brand resolution.
+ */
+
 /** Gets the last ThoughtId or value in a Path or Context. */
-const head = <T>(list: T[]) => list[list.length - 1]
+function head(list: SimplePath): ThoughtId
+function head<T>(list: T[]): T
+// eslint-disable-next-line jsdoc/require-jsdoc
```

Comment:
For overloaded functions, just move the JSDOC above the implementation.

cf00a0321e